src/protos/pushers.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+/*************************************************/
+/******** Main messages that will be sent ********/
+/*************************************************/
+
+// Configuration details that may or may not require a reboot to implement
+message PusherConfig {
+    PusherType type = 1;
+    int32 ExtendPin = 2;
+    bool CanBrake = 3;
+    // Can be skipped if CanBrake == false
+    int32 BrakePin = 4;
```

Comment:
Are you putting a full bridge on there or something?

Reply:
What are the 4 pins we'll have?

Are Duty Cycle and PWM Frequency not linked? And should we be exposing those to the user or just keeping that internal to the Detlaff code with a couple human parameters?

Reply:
Yeah it's a full bridge, so you will have one pin to control each FET. Duty cycle is the PWM percentage, not really linked to PWM frequency. I think your parameters would be pusher voltage which is user adjustable, and dead time and PWM frequency which are hidden in an advanced settings page in the web app.

Reply:
Okay, so we need 4 pins for forward/backwards high/low, duty cycle, dead time, and pwm values?

Reply:
I meant

pusherHigh1Pin
pusherLow1Pin
pusherHigh2Pin
pusherLow2Pin
pusgerTargetVoltage_mv
pusherDeadTime_ms
pusherPwmFreq_hz

And if we don't have a reverse direction boolean we need that too

Reply:
I'm going to nix the pins and dead time, per the comments above, assuming that we'll store them in the board config.

Can you explain a little more how/why a user would want to control the TargetVoltage and PWM freq?

Reply:
Ok sounds good.

TargetVoltage is for controlling pusher speed, or to reduce pusher current to keep it from burning out. Say you have a 2S rated pusher motor and you're running 4S for your flywheels, you could set TargetVoltage to 8V and it would calculate duty cycle = pusherTargetVoltage / batteryVoltage to keep the RoF constant.

PWM Frequency though you're probably right that the user probably won't need to modify. There's a tradeoff because you get better motor driver efficiency at lower frequency, but then it gets into the audible range. Hopefully the pusher driver will be beefy enough that we can take the efficiency hit and always run it outside of audible frequency.